core/capabilities/syncer.go

```diff
@@ -143,38 +170,49 @@ func (s *registrySyncer) Start(ctx context.Context) error {
 			s.subServices = append(s.subServices, localTrigger)
 			s.subServices = append(s.subServices, mockMercuryDataProducer)
 			// ----
-		}
+		}*/
 
-		underlying, err2 := s.registry.GetTrigger(ctx, capId)
-		if err2 != nil {
-			// NOTE: it's possible that the jobs are not launched yet at this moment.
-			// If not found yet, Syncer won't add to Registry but retry on the next tick.
-			return err2
-		}
-		workflowDONs := map[string]capabilities.DON{
-			workflowDonInfo.ID: workflowDonInfo,
-		}
-		triggerCap := remote.NewTriggerPublisher(config, underlying, triggerInfo, triggerCapabilityDonInfo, workflowDONs, s.dispatcher, s.lggr)
-		err = s.dispatcher.SetReceiver(capId, triggerCapabilityDonInfo.ID, triggerCap)
-		if err != nil {
-			s.lggr.Errorw("capability DON failed to set receiver", "capabilityId", capId, "donId", triggerCapabilityDonInfo.ID, "error", err)
-			return err
+		count := 0
+		for {
+			count++
+			if count > maxRetrySeconds {
+				s.lggr.Error("failed to get Streams Trigger from the Registry")
+				return
+			}
+			underlying, err2 := s.registry.GetTrigger(ctx, capId)
+			if err2 != nil {
+				// NOTE: it's possible that the jobs are not launched yet at this moment.
+				// If not found yet, Syncer won't add to Registry but retry on the next tick.
+				s.lggr.Infow("trigger not found yet ...", "capabilityId", capId, "error", err)
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			workflowDONs := map[string]capabilities.DON{
+				workflowDonInfo.ID: workflowDonInfo,
+			}
+			triggerCap := remote.NewTriggerPublisher(config, underlying, triggerInfo, triggerCapabilityDonInfo, workflowDONs, s.dispatcher, s.lggr)
+			err = s.dispatcher.SetReceiver(capId, triggerCapabilityDonInfo.ID, triggerCap)
+			if err != nil {
+				s.lggr.Errorw("capability DON failed to set receiver", "capabilityId", capId, "donId", triggerCapabilityDonInfo.ID, "error", err)
+				return
+			}
+			s.subServices = append(s.subServices, triggerCap)
+			break
```

Comment:
Would this be cleaner using a backoff.Backoff with permanent retry?

Reply:
yes but probably not worth it - again this Syncer won't be used in prod.